src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -34,221 +36,152 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
-        if (index != 0)
-            return index;
-        else if (intersection.size() == 3 &&
-                 intersection[1].turn.instruction.type == TurnType::Fork)
-        {
-            // Forks themselves do not contain a `obvious` turn index. If we look at a fork that has
-            // a one-sided sliproad, however, the non-sliproad can be considered `obvious`. Here we
-            // assume that this could be the case and check for a potential sliproad/non-sliproad
-            // situation.
-            NodeID intersection_node_one = SPECIAL_NODEID, intersection_node_two = SPECIAL_NODEID;
-            const auto intersection_following_index_one = findNextIntersectionForRoad(
-                intersection_node_id, intersection[1], intersection_node_one);
-            const auto intersection_following_index_two = findNextIntersectionForRoad(
-                intersection_node_id, intersection[2], intersection_node_two);
-            // in case of broken roads, we return
-            if (intersection_following_index_one.empty() ||
-                intersection_following_index_two.empty())
-                return 0;
-
-            // In case of loops at the end of the road, we will arrive back at the intersection
-            // itself. If that is the case, the road is obviously not a sliproad.
-
-            // a sliproad has to enter a road without choice
-            const auto couldBeSliproad = [](const Intersection &intersection) {
-                if (intersection.size() != 3)
-                    return false;
-                if ((intersection[1].entry_allowed && intersection[2].entry_allowed) ||
-                    intersection[0].entry_allowed)
-                    return false;
-                return true;
-            };
-
-            if (couldBeSliproad(intersection_following_index_one) &&
-                intersection_node_id != intersection_node_two)
-                return 2;
-            else if (couldBeSliproad(intersection_following_index_two) &&
-                     intersection_node_id != intersection_node_one)
-                return 1;
-            else
-                return 0;
-        }
-        else
-            return 0;
-    }();
+    // Potential non-sliproad road (bc), leading to the intersection (c) the Sliproad (bd) shortcuts
+    const auto &next_road = intersection[*obvious];
+    const auto next_road_edge_data = node_based_graph.GetEdgeData(next_road.turn.eid);
 
-    if (obvious_turn_index == 0)
-        return intersection;
+    const auto source_edge_data = node_based_graph.GetEdgeData(source_edge_id);
 
-    const auto &next_road = intersection[obvious_turn_index];
+    // The road leading to the intersection (bc) has to continue from our source
+    {
+        auto same_road_category =
+            next_road_edge_data.road_classification == source_edge_data.road_classification;
+        auto same_name = next_road_edge_data.name_id != EMPTY_NAMEID &&
```

Comment:
Yeah this will be done in https://github.com/Project-OSRM/osrm-backend/pull/2945 - I just haven't rebased and finished it yet.

There will be some conflicts, for sure.